### ISSUE TITLE
[HTML] tag names don't comply with HTML5 specification

### ISSUE THREAD

**Issue by deathaxe** [sublimehq/Packages#1619]
According to the html5 specification nearly any character is allowed to be part of a tag name.

see: https://html.spec.whatwg.org/multipage/parsing.html#tag-name-state

The current HTML implementation does not comply with the specification as 

1. it only allows lowercase ASCII letters, numbers and few punctuation `[a-z_][a-z0-9:_]*-[a-z0-9:_-]+`
2. it doesn't respect uppercase (custom-) tag names being converted lowercase implicitly and therefore are valid tag names.
3. it fails to scope custom tag names like `form@custom` correctly.

Examples of valid tag names:

```html
<form@custom attribute="value" />

<FORM-CUSTOM attribute="value" />

```

P.S.: Github does not, too.  ;-)

**Comment by user-9826342** [comment]
I feel that I need to leave a comment here for anyone who may say *"but such syntax doesn't appear in real code"*. One of the most popular websites, YouTube, uses the `-` in almost every tag name:

<details>
<summary>YouTube specific tag names</summary>

```
app-drawer
array-selector
iron-iconset-svg
iron-list
iron-media-query
paper-button
paper-dialog
paper-spinner-lite
paper-tooltip
yt-activity-manager
yt-app-context
yt-auth-manager
yt-formatted-string
yt-gfeedback-manager
yt-guide-manager
yt-history-manager
yt-hotkey-manager
yt-icon
yt-icon-button
yt-img-shadow
yt-mdx-manager
yt-navigation-manager
yt-network-manager
yt-page-navigation-progress
yt-player-manager
yt-playlist-manager
yt-service
yt-view-count-renderer
yt-visibility-monitor
ytd-app
ytd-badge-supported-renderer
ytd-comments
ytd-compact-video-renderer
ytd-companion-legal-text-renderer
ytd-companion-slot-renderer
ytd-debug-menu
ytd-expander
ytd-horizontal-card-list-renderer
ytd-item-section-renderer
ytd-masthead
ytd-menu-renderer
ytd-message-renderer
ytd-metadata-row-container-renderer
ytd-metadata-row-renderer
ytd-miniplayer
ytd-notification-topbar-button-renderer
ytd-page-manager
ytd-pla-shelf-renderer
ytd-playability-error-supported-renderers
ytd-player
ytd-player-legacy-desktop-watch-ads-renderer
ytd-playlist-panel-renderer
ytd-playlist-panel-video-renderer
ytd-popup-container
ytd-report-form-loader
ytd-searchbox
ytd-sentiment-bar-renderer
ytd-subscribe-button-renderer
ytd-third-party-manager
ytd-thumbnail
ytd-thumbnail-overlay-time-status-renderer
ytd-toggle-button-renderer
ytd-topbar-logo-renderer
ytd-topbar-menu-button-renderer
ytd-transcript-loader
ytd-video-meta-block
ytd-video-owner-renderer
ytd-video-primary-info-renderer
ytd-video-quality-promo-renderer
ytd-video-secondary-info-renderer
ytd-watch
ytd-watch-next-secondary-results-renderer
ytd-yoodle-renderer
ytd-youthere-manager
```

</details><br/>

Though I didn't see `form@custom` anywhere, but I think it would be nice to support it too.

**Comment by deathaxe** [comment]
Changes to fix this issue would conflict with PR #1566 and #1567. Therefore those should be merged first.

**Comment by infininight** [comment]
Read the spec on [custom element names](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-elements-core-concepts), specifically the part on PotentialCustomElementName, they must contain a `-` character and must not be a short list of specific ones for legacy reasons. That's what all the YouTube ones contain `-` and `<form@custom>` wouldn't be valid.
